File: backend/src/routes/health.ts
import type { FastifyInstance } from "fastify";

import { LOCAL_CONSOLE_INGEST_SCHEMA_VERSION } from "../../../shared/src/enums.js";

export function registerHealthRoutes(app: FastifyInstance): void {
  app.get("/health", async () => ({
    ok: true,
    serverTimeMs: Date.now(),
    schemaVersion: LOCAL_CONSOLE_INGEST_SCHEMA_VERSION,
  }));
}

File: backend/internal/routes/health.go
// Package routes mirrors backend/src/routes.
package routes

import (
	"time"

	"github.com/gin-gonic/gin"
)

// ingest schema version kept in lockstep with shared/src/enums.ts.
const ingestSchemaVersion = "lynx-server.ingest.v1"

// RegisterHealth mirrors registerHealthRoutes.
func RegisterHealth(router gin.IRoutes) {
	router.GET("/health", func(c *gin.Context) {
		c.JSON(200, map[string]any{
			"ok":            true,
			"serverTimeMs":  time.Now().UnixMilli(),
			"schemaVersion": ingestSchemaVersion,
		})
	})
}
